lib/screens/pokemon_info/widgets/tab_base_stats.dart

```diff
@@ -112,6 +115,39 @@ class _PokemonBaseStatsState extends State<PokemonBaseStats> with SingleTickerPr
     ];
   }
 
+  String _removeTrailingZero(double n){
+    String s = n.toString().replaceAll(RegExp(r"([.]*0)(?!.*\d)"), "");
+    return s;
+  }
+
+  double _getTypeEffectiveness(Pokemon pokemon, String type){
+    double effectiveness = 1;
+    for(String pokemonType in pokemon.types){
+      Type ty = getTypeFromString(pokemonType);
+      if(ty.immune.contains(type)){
+        return 0;
+      }
+      if(ty.superEffective.contains(type)){
+        effectiveness *= 2;
+      }
+      if(ty.notEffective.contains(type)){
+        effectiveness *= 0.5;
+      }
+    }
+    return effectiveness;
+  }
+
+  List<Widget> generateEffectivenessWidget(Pokemon pokemon){
+    Map<String, double> effectiveness = Map.fromIterable(listOfTypes, key: (type) => type, value: (type) => _getTypeEffectiveness(pokemon, type));
```

Comment:
You can put this inside the `_getTypeEffectiveness` and only need to pass `pokemon` so the method will become  `Map<String, double> _getTypeEffectiveness(pokemon)`. Inside this method, you can use nested `for` loop instead of  `Map.fromIterable`